### DISCUSSION TITLE
Using SU2's Adjoint Optimization For Inverse Problem

### DISCUSSION THREAD

**Discussion by Gaurav11ME** [su2code/SU2#1624] [Q&A]
I am exploring the use of SU2's adjoint capability for solving an inverse problem. I wish to know if it is possible to use this to optimize a spatially varying parameter (e.g., spatially varying viscosity). I have measured values of the velocity field in the flow domain, and below is the process I wish to follow for the inference.
1. Guess viscosity values at grid points.
2. Run a RANS solution
3. Compare the RANS velocity field with the measured field
4. Use loss function = (u_rans-u_measure)^2
5. Use adjoint optimization to get the updated viscosity field
6. Repeat 2-5 until convergence (Loss<set_threshold)

**Comment by pcarruscag** [comment]
Yes, this is possible, but you would need to write code for:
Reading the viscosity distribution from somewhere
Using it in the solver
Computing the new objective function
Extracting derivatives wrt new variables

You can have a look at CElasticitySolver, it does similar actions to what I described.

**Reply by Gaurav11ME** [reply]
Thank you for the quick response. I have figured out how to add a configuration option for reading data from a file. I will check CElasticitySolver.

**Reply by Gaurav11ME** [reply]
I could not locate CElasticitySolver file. I found CFEASolver. Are they the same?

**Comment by bigfooted** [comment]
I recommend working from Visual Studio Code  [](https://code.visualstudio.com/), and then search for "Elasticity" to see files associated with this solver, for instance the files in the directory _src/numerics/elasticity_

**Reply by Gaurav11ME** [reply]
Ok. I will try that.
